infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -493,80 +408,68 @@ private void registerToLogReplicationLock() {
      * as source (sender/producer) or sink (receiver).
      */
     private void onLeadershipAcquire() {
-        switch (localClusterDescriptor.getRole()) {
+        switch (topologyDescriptor.getLocalClusterDescriptor().getRole()) {
             case SOURCE:
                 log.info("Start as Source (sender/replicator)");
-                if (replicationManager == null) {
-                    replicationManager = new CorfuReplicationManager(replicationContext, localNodeDescriptor,
-                        remoteSessionToMetadataManagerMap, serverContext.getPluginConfigFilePath(),
-                            getCorfuRuntime(), upgradeManager);
-                } else {
-                    // Replication Context contains the topology which
-                    // must be updated if it has changed
-                    replicationManager.setReplicationContext(replicationContext);
-                }
-
-                // Start log replication manager for each Sink
-                for (ClusterDescriptor remoteCluster : topologyDescriptor.getSinkClusters().values()) {
-                    replicationManager.start(remoteCluster, replicationContext);
-                }
+                sessionManager.startReplication();
 
                 // Set initial/default replication status for newly added Sink clusters
-                initReplicationStatusForRemoteClusters(true);
-                lockAcquireSample = recordLockAcquire(localClusterDescriptor.getRole());
-                processCountOnLockAcquire(localClusterDescriptor.getRole());
+                // TODO[Anny]: I don't think this is needed / confirm
+                // initReplicationStatusForRemoteClusters(true);
```

Comment:
I agree that this is not required as a default status is populated on session creation.